pkg/kubectl/resource/result.go

```diff
@@ -268,7 +278,8 @@ func AsVersionedObjects(infos []*Info, version schema.GroupVersion, encoder runt
 		// objects that are not part of api.Scheme must be converted to JSON
 		// TODO: convert to map[string]interface{}, attach to runtime.Unknown?
 		if !version.Empty() {
-			if _, _, err := api.Scheme.ObjectKinds(info.Object); runtime.IsNotRegisteredError(err) {
+			var o runtime.Object = info.Object
+			if _, _, err := scheme.Scheme.ObjectKinds(o); runtime.IsNotRegisteredError(err) {
```

Comment:
Same here. If `o` is of internal version, `scheme.Scheme` won't help.

Reply:
agree... AsVersionedObjects implies there is question as to whether the input objects are internal or of a different version... either implies conversion, which an external-only scheme won't be able to help with